vividus-analytics/src/main/java/org/vividus/analytics/GoogleAnalyticsFacade.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.http.message.BasicNameValuePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.analytics.model.AnalyticsEvent;
+import org.vividus.http.HttpMethod;
+import org.vividus.http.client.IHttpClient;
+
+public class GoogleAnalyticsFacade
+{
+    private static final String CLIENT_ID = "cid";
+
+    private static final String TRACKING_ID = "tid";
+
+    private static final String HIT_TYPE = "t";
+
+    private static final String MEASUREMENT_API_VERSION = "v";
+
+    private static final Pattern PROJECT_NAME_PATTERN = Pattern.compile("(.+)(-\\d.+)?(\\\\|/)?");
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GoogleAnalyticsFacade.class);
+
+    private URI analyticsUri;
+    private String trackingId;
+    private List<NameValuePair> defaultParameters;
+    private final IHttpClient httpClient;
+
+    public GoogleAnalyticsFacade(IHttpClient httpClient)
+    {
+        this.httpClient = httpClient;
+    }
+
+    public void init()
+    {
+        defaultParameters = new ArrayList<>();
+        defaultParameters.add(pairOf(MEASUREMENT_API_VERSION, "1"));
+        defaultParameters.add(pairOf(HIT_TYPE, "event"));
+        defaultParameters.add(pairOf(TRACKING_ID, trackingId));
+        defaultParameters.add(pairOf(CLIENT_ID, cid()));
+    }
+
+    private static String cid()
+    {
+        String[] pathParts = System.getProperty("user.dir").split("\\\\|/");
+        String projectName = pathParts[pathParts.length - 1];
+        if ("scripts".equals(projectName))
+        {
+            projectName = PROJECT_NAME_PATTERN.matcher(pathParts[pathParts.length - 2]).replaceFirst("$1");
+        }
+        return UUID.nameUUIDFromBytes(DigestUtils.sha512Hex(projectName).getBytes(StandardCharsets.UTF_8)).toString();
+    }
+
+    public void postEvent(AnalyticsEvent analyticsEvent)
+    {
+        List<NameValuePair> params = new ArrayList<>();
+        params.addAll(defaultParameters);
+        params.addAll(convertToNameValuePairs(analyticsEvent));
+        StringEntity entity = new StringEntity(URLEncodedUtils.format(params, UTF_8),
+                UTF_8);
+        post(entity);
+    }
+
+    private List<BasicNameValuePair> convertToNameValuePairs(AnalyticsEvent analyticsEvent)
+    {
+        return analyticsEvent.getPayload()
+                             .entrySet()
+                             .stream()
+                             .map(e -> pairOf(e.getKey(), e.getValue()))
+                             .collect(Collectors.toList());
+    }
+
+    private BasicNameValuePair pairOf(String key, String value)
+    {
+        return new BasicNameValuePair(key, value);
+    }
+
+    private void post(HttpEntity entity)
+    {
+        try
+        {
+            HttpEntityEnclosingRequestBase createEntityEnclosingRequest = HttpMethod.POST
+                    .createEntityEnclosingRequest(analyticsUri, entity);
+            createEntityEnclosingRequest.setHeader("User-Agent", "");
```

Comment:
just interesting why is it needed?

Reply:
1. It looks like analytics is missing some events with our User-Agent
2. It is used to calculate statistics we are no interested in;

Reply:
Just in case we could send User-Agent via Measurement Protocol and it will be used to calculate what browser is used, but we track profiles not the browsers;